third-party/realdds/include/realdds/dds-stream-profile.h

```diff
@@ -87,6 +94,10 @@ class dds_stream_profile
     }
 
 public:
+    std::shared_ptr< dds_stream_base > stream() const { return _stream.lock(); }
+    // This is for initialization and is called from dds_stream_base only!
+    void init_stream( std::weak_ptr< dds_stream_base > const & stream );

```

Comment:
What does it initialize? Can call it `set_stream`

Reply:
It sets the "owner" of the profile.
This is so you can do `profile->stream()->name()` for example. I.e., by passing profiles around, you can then get their streams (for example, for the `open()` call)

Reply:
Didn't want to call it `set` -- wanted to communicate that it's a one-time thing.

Reply:
Same for `set_profiles`, by the way -- but thought that, over there, set might be better. What do you think, `set_profiles` or `init_profiles`?

Reply:
How about `dds_stream_profile::associate_with_stream` and `dds_stream::set_supported_profiles`?

Reply:
Changed to `init_profiles`.
I don't want any new terminology. Either `set` or `init`. The `supported` is unneeded - do we have unsupported profiles?